sympy/series/tests/test_limits.py

```diff
@@ -372,10 +372,10 @@ def test_issue_4547():
 
 def test_issue_5164():
     assert limit(x**0.5, x, oo) == oo**0.5 is oo
-    assert limit(x**0.5, x, 16) == S(16)**0.5
+    assert limit(x**0.5, x, 16) == 4 # Should this be a float?
```

Comment:
I believe so, why would it lose its "floatness"?

Reply:
Probably because the float is converted into (or interpreted as) a Rational somewhere.

Reply:
I reverted this change